text/0000-external-doc-attribute.md

```diff
@@ -0,0 +1,210 @@
+- Feature Name: external_doc
+- Start Date: 2017-04-26
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Documentation is an important part of any project, it allows developers to
+explain how to use items within a library as well as communicate the intent of
+how to use it through examples. Rust has long championed this feature through
+the use of documentation comments and `rustdoc` to generate beautiful, easy to
+navigate documentation. However, there is no way right now to have documentation
+be imported into the code from an external file. This RFC proposes a way to
+extend the functionality of Rust to include this ability.
+
+# Motivation
+[motivation]: #motivation
+
+1. Many smaller crates are able to do all of the documentation that's needed in
+   a README file within their repo. Being able to include this as a crate or
+   module level doc comment would mean not having to duplicate documentation and
+   is easier to maintain. This means that one could run `cargo doc` with the
+   small crate as a dependency and be able to access the contents of the README
+   without needing to go online to the repo to read it. This also would help
+   with [this issue on
+   crates.io](https://github.com/rust-lang/crates.io/issues.81) by making it
+   easy to have the README in the crate and the crate root at the same.
+2. The feature would provide a way to have easier to read code for library
+   maintainers. Sometimes doc comments are quite long in terms of line count
+   (items in
+   [libstd](https://github.com/rust-lang/rust/blob/master/src/libstd) are a good
+   example of this). Doc comments document behavior of functions, structs, and
+   types to the end user, they do not explain for a coder working on the library
+   as to how they work internally. When actually writing code for a
+   library the doc comments end up cluttering the source code making it harder
+   to find relevant lines to change or skim through and read what is going on.
+3. Localization is something else that would further open up access to the
+   community. By providing docs in different languages we could significantly
+   expand our reach as a community and be more inclusive of those where English
+   is not their first language. This would be made possible with a config flag
+   choosing what file to import as a doc comment.
+
+These are just a few reasons as to why we should do this, but the expected
+outcome of this feature is expected to be positive with little to no downside
+for a user.
+
+# Detailed design
+[design]: #detailed-design
+
+All files included through the attribute will be relative paths from the crate
+root directory. Given a file like this stored in `docs/example.md`:
+
+```md
+# I'm an example
+This is a markdown file that gets imported to Rust as a Doc comment.
+```
+
+And code like this:
+
+```rust
+#[external_doc("docs/example.md", "line")]
```

Comment:
This seems like a better idea. I'll update the RFC when I can with this syntax as I find it more ergonomic and simmilar to what already exists in Rust.